plugins/inputs/tacacs/tacacs.go

```diff
@@ -0,0 +1,164 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package tacacs
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/nwaples/tacplus"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Tacacs struct {
+	Servers         []string        `toml:"servers"`
+	Username        config.Secret   `toml:"username"`
+	Password        config.Secret   `toml:"password"`
+	Secret          config.Secret   `toml:"secret"`
+	RemAddr         string          `toml:"remaddr"`
+	ResponseTimeout config.Duration `toml:"response_timeout"`
+	Log             telegraf.Logger `toml:"-"`
+}
+
+//go:embed sample.conf
+var sampleConfig string
+
+func (t *Tacacs) SampleConfig() string {
+	return sampleConfig
+}
+
+func (t *Tacacs) Init() error {
+	if len(t.Servers) == 0 {
+		t.Servers = []string{"127.0.0.1"}
+	}
+	return nil
```

Comment:
Maybe also check for other empty required fields? I think `RemAddr` and credentials need to be set?

Reply:
> mandatory fields are uncommented in the sample conf. If not set, I think these should catch it:
> 
> ```
> 	username, err := t.Username.Get()
> 	if err != nil {
> 		return fmt.Errorf("getting username failed: %v", err)
> 	}
> ```

The problem is that these only get checked at runtime and result in an error each interval.

> The other non mandatory are now defaults: `return &Tacacs{RemAddr: "127.0.0.1", ResponseTimeout: config.Duration(time.Second * 5)}`

What if someone sets `request_ip = ""` or an invalid IP? A valid config still needs to be checked and alerted in `Init()`.

Reply:
Im adding check for the remaining items (username, password, secret). Although this will fail on users that may want to use empty string for those fields (as empty string is zero bytes, which is considered same as not set by the Empty() check of the config.Secret type )